src/main/java/reposense/report/RepoCloner.java

```diff
@@ -104,8 +102,8 @@ private boolean spawnCloneProcess(String outputPath, RepoConfiguration config) t
             Files.createDirectories(rootPath);
 
             logger.info(String.format(MESSAGE_START_CLONING, config.getLocation()));
-            crp = runCommandAsync(rootPath, GitClone.getCloneBareCommand(config,
-                    FileUtil.getBareRepoFolderName(config)));
+            crp = GitClone.cloneBareAsync(config, rootPath, FileUtil.getBareRepoFolderName(config));
+
         } catch (RuntimeException | IOException e) {
```

Comment:
The RuntimeException can only be produced by the `runCommandAsync` previously right?
In that case, make `cloneBareAsync` handle that exception and throw it back out as a clone exception.